credentials/google/google.go

```diff
@@ -35,63 +35,56 @@ const tokenRequestTimeout = 30 * time.Second
-// with google services.
+// DefaultCredentialsOptions constructs options to build DefaultCredentials.
+type DefaultCredentialsOptions struct {
+	// PerRPCCreds is a  per RPC credentials that is passed to a bundle.
+	PerRPCCreds credentials.PerRPCCredentials
+}
+
+// NewDefaultCredentialsWithOptions returns a credentials bundle that is
+// configured to work with google services.
 //
 // This API is experimental.
-func NewDefaultCredentials() credentials.Bundle {
+func NewDefaultCredentialsWithOptions(opts DefaultCredentialsOptions) credentials.Bundle {
+	perRPC := opts.PerRPCCreds
+	if perRPC == nil {
+		ctx, cancel := context.WithTimeout(context.Background(), tokenRequestTimeout)
+		defer cancel()
+		var err error
+		perRPC, err = oauth.NewApplicationDefault(ctx)
+		if err != nil {
+			logger.Warningf("google default creds: failed to create application oauth: %v", err)
+		}
+	}
 	c := &creds{
 		newPerRPCCreds: func() credentials.PerRPCCredentials {
-			ctx, cancel := context.WithTimeout(context.Background(), tokenRequestTimeout)
-			defer cancel()
-			perRPCCreds, err := oauth.NewApplicationDefault(ctx)
-			if err != nil {
-				logger.Warningf("google default creds: failed to create application oauth: %v", err)
-			}
-			return perRPCCreds
+			return perRPC
 		},
 	}
 	bundle, err := c.NewWithMode(internal.CredsBundleModeFallback)
 	if err != nil {
-		logger.Warningf("google default creds: failed to create new creds: %v", err)
+		logger.Warningf("compute engine creds with per rpc: failed to create new creds: %v", err)
 	}
 	return bundle
 }
 
-// NewComputeEngineCredentials returns a credentials bundle that is configured to work
-// with google services. This API must only be used when running on GCE. Authentication configured
-// by this API represents the GCE VM's default service account.
+// NewDefaultCredentials returns a credentials bundle that is configured to work
+// with google services.
 //
 // This API is experimental.
-func NewComputeEngineCredentials() credentials.Bundle {
-	return NewComputeEngineCredsWithOptions(ComputeEngineCredsOptions{})
-}
-
-// ComputeEngineCredsOptions constructs compite engine credentials with options.
-type ComputeEngineCredsOptions struct {
-	// PerRPCCreds is a  per RPC credentials that is passed to a bundle.
-	PerRPCCreds credentials.PerRPCCredentials
+func NewDefaultCredentials() credentials.Bundle {
+	return NewDefaultCredentialsWithOptions(DefaultCredentialsOptions{})
 }
 
-// NewComputeEngineCredsWithOptions returns a credentials bundle that is configured to work
-// with google services. This API must only be used when running on GCE.
+// NewComputeEngineCredentials returns a credentials bundle that is configured to work
+// with google services. This API must only be used when running on GCE. Authentication configured
+// by this API represents the GCE VM's default service account.
 //
 // This API is experimental.
```

Comment:
From the other PR:

> ComputeEngineCredentials should be deprecated.

Should this be updated to say it's deprecated and use default credentials instead?

Reply:
Nothing needs to be deprecated. This function still serves its purpose.
But I guess the only user (cloud libraries) will stop using it after this PR.